lib/rets/client.rb

```diff
@@ -105,9 +105,11 @@ def find_with_given_retry(retries, opts)
     end
 
     def handle_find_failure(retries, opts, e)
-      if retries < opts.fetch(:max_retries, 3)
+      max_retries = opts.fetch(:max_retries, options.fetch(:max_retries, 3))
```

Comment:
`opts` and `options` both being used here is a little hard to read, might be worth changing the name of the local variable (`local_options`?), but I'm not too worried about it.